packages/amplify-headless-interface/src/interface/api/add.ts

```diff
@@ -18,18 +18,43 @@ export interface AppSyncServiceConfiguration {
    * Additional methods of authenticating API requests
    */
   additionalAuthTypes?: AppSyncAuthType[];
-  conflictResolution?: {
-    resolutionStrategy: ResolutionStrategy;
-  };
+  conflictResolution?: ConflictResolution;
 }
 
-export enum ResolutionStrategy {
-  OPTIMISTIC_CONCURRENCY = 'OPTIMISTIC_CONCURRENCY',
-  LAMBDA = 'LAMBDA',
-  AUTOMERGE = 'AUTOMERGE',
-  NONE = 'NONE',
+export interface ConflictResolution {
+  defaultResolutionStrategy?: ResolutionStrategy;
+  perModelResolutionStrategy?: PerModelResolutionstrategy[];
 }
 
+export interface PerModelResolutionstrategy {
+  resolutionStrategy: ResolutionStrategy;
+  entityName: string;
+}
+
+export interface PredefinedResolutionStrategy {
```

Comment:
Why have we separated Lambda resolution strategy out from the other resolution strategies?